src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -1316,6 +1335,7 @@ internal class HeaderStoreItemInfo
             internal object? RawValue { get; set; }
             internal object? InvalidValue { get; set; }
             internal object? ParsedValue { get; set; }
+            internal object? OriginalRawValue { get; set; }
```

Comment:
This increases the size of every HeaderStoreItemInfo by 20%. Why is that ok?

Reply:
I realize that, and initially tried to remove `RawValue` resets and keep track everything there, but it proved to be extremely complex and dangerous because it requires changing the parsing logic in multiple places.